java/bulk-export/bulk-export-task-execution/src/main/java/sleeper/bulkexport/taskexecution/ECSBulkExportTaskRunner.java

```
@@ -105,29 +105,41 @@ public static void main(String[] args) throws ObjectFactoryException, IOExceptio
+     * @param  dynamoDBClient          a DynamoDB client
+     * @param  hadoopConf              a Hadoop configuration
+     * @throws IOException             if there is an error interacting with S3
+     * @throws IllegalStateException   if there is an error receiving the message and deseralising it
      */
     public static void runECSBulkExportTaskRunner(
             InstanceProperties instanceProperties, TablePropertiesProvider tablePropertiesProvider,
-            SqsClient sqsClient, S3Client s3Client, DynamoDbClient dynamoDBClient, Configuration hadoopConf) throws IOException, IteratorCreationException, ObjectFactoryException {
+            SqsClient sqsClient, S3Client s3Client, DynamoDbClient dynamoDBClient,
+            Configuration hadoopConf) throws IOException, IllegalStateException {
         SqsBulkExportQueueHandler exportQueueHandler = new SqsBulkExportQueueHandler(sqsClient,
                 tablePropertiesProvider, instanceProperties);
         LOGGER.info("Waiting for leaf partition bulk export job from queue {}",
                 instanceProperties.get(CdkDefinedInstanceProperty.LEAF_PARTITION_BULK_EXPORT_QUEUE_URL));
+        Optional<SqsMessageHandle> messageHandleOpt;
+
+        try {
+            messageHandleOpt = exportQueueHandler.receiveMessage();
+        } catch (IOException | IllegalStateException e) {
+            LOGGER.error("There is a problem getting the message.", e);
+            throw e;
+        }
 
-        Optional<SqsMessageHandle> messageHandleOpt = exportQueueHandler.receiveMessage();
         if (messageHandleOpt.isPresent()) {
             SqsMessageHandle messageHandle = messageHandleOpt.get();
             try {
-                BulkExportLeafPartitionQuery exportTask = messageHandle.getJob();
+                BulkExportLeafPartitionQuery exportTask;
+                if (messageHandle.getJob() == null) {
+                    throw new NullPointerException("No job found in bulk export message");
```

This check isn't needed as the job can never be null. If it was null an exception would have been thrown inside SqsBulkExportQueueHandler.

Also, I'm not sure there's a benefit in general to explicitly throwing a null pointer exception like this, rather than implicitly due to a null pointer dereference.

---

Removed